src/Angular.js

```diff
@@ -1129,9 +1129,9 @@ function parseKeyValue(/**string*/keyValue) {
   forEach((keyValue || "").split('&'), function(keyValue) {
     if ( keyValue ) {
       key_value = keyValue.replace(/\+/g,'%20').split('=');
```

Comment:
I know it is slightly more expensive, but you can save some of the weirdness later in the code if by changing `.split('=')` with `.split(/=(.*)/, 2)`